tpot/tpot.py

```diff
@@ -1068,6 +1072,39 @@ def _select_fwe(self, input_df, alpha):
         mask_cols = list(training_features.iloc[:, mask].columns) + self.non_feature_columns
         return input_df[mask_cols].copy()
 
+    def _ekf(self, input_df, ekf_index, k_best):
+        """Uses expert knowledge source to subset/mask columns of the input data
+        
+        Parameters
+        ----------
+        input_df: pandas.DataFrame {n_samples, n_features+['class', 'group', 'guess']}
+            Input DataFrame to perform feature selecton on
+        expert_source: Boolean list/mask for features provided by user
+        
+        Returns
+        -------
+        subsetted_df: andas.DataFrame {n_samples, n_filtered_features + ['guess', 'group', 'class']}
+            Returns a DataFrame containing the 'best' features provided by expert_source
+        
+        """
+#        training_features = input_df.loc[input_df['group'] == 'training'].drop(self.non_feature_columns, axis=1)
+#        all_ekf_cols = len(training_features.columns)
+        training_features = input_df
+        all_ekf_cols = training_features.shape[1]
+    
+        if k_best < 1:
+            k_best = 1
+        elif k_best >= all_ekf_cols:
+            k_best = all_ekf_cols        
+        
+        if self.expert_source[ekf_index].dtype==int:
```

Comment:
Make this check something like `set(self.expert_source[ekf_index]) == set([True, False])` to guarantee that we're using a mask here.